### PR TITLE
Specify return type of DateTime::createFromImmutable() and DateTimeImmutable::createFromMutable()

### PR DESCRIPTION
**Pull request by kocsismate**

### CODE REVIEW

File: reference/datetime/datetime/createfromimmutable.xml [hunk 1]
```diff
@@ -9,7 +9,7 @@
  <refsect1 role="description">
   &reftitle.description;
   <methodsynopsis>
-   <modifier>public</modifier> <modifier>static</modifier> <type>DateTime</type><methodname>DateTime::createFromImmutable</methodname>
+   <modifier>public</modifier> <modifier>static</modifier> <type>static</type><methodname>DateTime::createFromImmutable</methodname>
```
Comment (kocsismate) [inline reference/datetime/datetime/createfromimmutable.xml:12]: I'm not sure if/how the return type description below needs any clarification.

**Comment by derickr** [comment]
I don't understand why you want to change this. It's more accurate I suppose, but it's much clearer for readers to know the exact type instead of having to then re-parse which class they're reading the docs for.

File: reference/datetime/datetime/createfromimmutable.xml (hunk 1 above)
Comment (cmb69) [reply reference/datetime/datetime/createfromimmutable.xml:12]: In my opinion, the return type section doesn't need to be changed; after all, what is returned is an instance of `DateTime(Immutable)` (possibly a subtype, but still an instance of the class).

However, this needs a changelog entry, doesn't it?

**Comment by cmb69** [comment]
> It's more accurate I suppose,

That. And accuracy looks like a good idea for technical documentation.

**Comment by kocsismate** [comment]
> I don't understand why you want to change this. It's more accurate I suppose, but it's much clearer for readers to know the exact type instead of having to then re-parse which class they're reading the docs for.

I'm changing this because `static` is the actual return type of these two methods, so as I'm currently syncing the manual with the stubs, this change came up.

In my opinion, showing the correct return type is vital, it's sometimes very important to know if a new DateTime(immutable), or the actually invoked class is going to be returned. Also, I think we should keep the docs in sync with the reality so that future (maybe more important) changes are much-much easier to document.

**Review by Girgias — APPROVED** [review]

File: reference/datetime/datetime/createfromimmutable.xml (hunk 1 above)
Comment (kocsismate) [reply reference/datetime/datetime/createfromimmutable.xml:12]: Oh yes! I missed that the behavior changed in PHP 8.0 via https://bugs.php.net/bug.php?id=60302

**Review by cmb69 — APPROVED** [review]
Thank you! The changelog entries are fine, as is the rest.

**Pull request merged**